### PR TITLE
Support cross-origin requests

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Allows any cross-origin request to be made against the API.

### CODE REVIEW

File: test/app.test.ts [hunk 1]
```diff
@@ -13,4 +13,11 @@ describe('the application', (): void => {
 
     expect(response.status).toEqual(404);
   });
+
+  it('should support cross-origin requests', async (): Promise<void> => {
+    const response = await request(app.callback()).get('/').set('Origin', 'http://example.com');
+
+    expect(response.get('Access-Control-Allow-Origin')).toBe('*');
+    expect(response.get('Vary')).toContain('Origin');
```
Comment (thewilkybarkid) [inline test/app.test.ts]: Need to check if this is an array, or still a comma-separated string.

File: src/app.ts [hunk 2]
```diff
@@ -6,6 +7,9 @@ const app = new Koa();
 const router = createRouter();
 
 app.use(logger());
+app.use(cors({
+  origin: '*',
```
Comment (stephenwf) [inline src/app.ts]: Was putting the origin as `*` meant to be a safe-guard against allowing cross-origin authorisation requests? If the store needs to deal with JWTs or any `Authorization` header values the `Access-Control-Allow-Origin` header _must_ match the `Origin` header from browsers. Setting the origin to this value is the default behaviour in koa cors (i.e. `app.use(cors())`). Just wanted to check if this value was on purpose (and if we should leave a comment?)

File: test/app.test.ts (hunk 1 above)
Comment (giorgiosironi) [reply test/app.test.ts]: It varies only on the presence or absence of `Origin` right?

File: src/app.ts (hunk 2 above)
Comment (thewilkybarkid) [reply src/app.ts]: Ah yeah, that's the point of `Vary: Origin`.

We will need authorization later down the line.

File: test/app.test.ts (hunk 1 above)
Comment (thewilkybarkid) [reply test/app.test.ts]: No, value of...

Comment (thewilkybarkid) [reply test/app.test.ts]: Relates to https://github.com/libero/article-store/pull/11#discussion_r345751382.

**Review by giorgiosironi — APPROVED** [review]

**Pull request merged**